### DISCUSSION TITLE
How to use UMP

### DISCUSSION THREAD

**Discussion by mga85** [Poing-Studios/Godot-AdMob-Android-iOS#112] [Q&A]
Hello,

Thanks a lot for the plug in.
I am trying to use UMP to show GDPR message in Android and IDFA message in IOS. In my AdMob account I configured both sections under Privacy & Messaging. 
In the main scene of my game, I call MobileAds.request_user_consent() function, in Android I can not see any GDPR relevant message. How can I use these messaging feature in plug in ? 

![image](https://user-images.githubusercontent.com/98719428/158230743-5099bc9b-5841-4de7-abf8-57ba99759c2f.png)

**Comment by gumaciel** [comment]
Hi! To see the GDPR Consent you need to be on EEA, if you are not on EEA, then activate this: 
![Captura de Tela 2022-03-18 às 19 40 26](https://user-images.githubusercontent.com/20030153/159093553-fe278cde-764f-47f8-9303-132d182e0f67.png)


Also: 

Add your [AdMob App ID](https://support.google.com/admob/answer/7356431) to your app's res://android/build/AndroidManifest.xml file by adding a <meta-data> tag with name com.google.android.gms.ads.APPLICATION_ID

**Reply by mga85** [reply]
Hi,

I am in EEA, and app is in production (in Android) but it does not show it, also IOS requires App Tracking Transparency related question to be shown but that does not pop up and Apple rejects the publication of the App (Guideline 2.1 - Information Needed) ? 

When I filled privacy policy in App Store Connect I used the suggestion from the site below for Admob. Based on this Apple requires showing the App Tracking Transparency request.

https://levelup.gitconnected.com/filling-out-the-app-privacy-section-in-app-store-connect-for-admob-users-bca0768ad86e

**Reply by gumaciel** [reply]
To show App Tracking Transparency request you need to use Funding Choices UI, read more here: https://support.google.com/fundingchoices/answer/9995402

**Reply by mga85** [reply]
Hello,

My funding choices are already active and I created two messages: one for IDFA and one for GDPR, as shown in the screenshot in the beginning. However, it is still not triggering any UMP message. In the first scene of the game I am calling MobileAds.request_user_consent() function.

**Reply by gumaciel** [reply]
Hi, strange because it should work. i will investigate this
